packages/utils/src/crypto.ts

```diff
@@ -6,6 +6,31 @@ export const SNARK_SCALAR_FIELD =
     '21888242871839275222246405745257275088548364400416034343698204186575808495617'
 export const F = BigInt(SNARK_SCALAR_FIELD)
 
+// Ordered merkle tree polysum constant
```

Comment:
Lots of these changes are wrong, the various `R` values should be removed, same with `genUserStateTransitionNullifier`, `genEpochNullifier`.

The implementation of `genStateTreeLeaf` and `genEpochTreeLeaf` should be reverted too

Reply:
Yes, you're taking the commits you made and applying them to a new base commit (the latest commit of `main`). Copying can work, but usually the only interactive parts of rebasing are handling conflicts. Copying code can work as long as it doesn't break your new functionality.